modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Tags/DefaultTagDefinitionStore.cs

```diff
@@ -0,0 +1,38 @@
+﻿using JetBrains.Annotations;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Volo.Abp;
+using Volo.Abp.DependencyInjection;
+
+namespace Volo.CmsKit.Tags
+{
+    public class DefaultTagDefinitionStore : ITagDefinitionStore, ITransientDependency
+    {
+        private readonly CmsKitOptions options;
+
+        public DefaultTagDefinitionStore(IOptions<CmsKitOptions> options)
+        {
+            this.options = options.Value;
+        }
+
+        public Task<TagDefiniton> GetTagDefinitionOrNullAsync([NotNull] string entityType)
+        {
+            Check.NotNullOrWhiteSpace(entityType, nameof(entityType));
+            return Task.FromResult(options.Tags.GetOrDefault(entityType));
+        }
+
+        public Task<List<TagDefiniton>> GetTagDefinitionsAsync()
+        {
+            return Task.FromResult(options.Tags.Values.ToList());
```

Comment:
Is there a reason why we implemented these methods are `async`? They are not `virtual`. We don't need to create an additional `Task` object here.

Reply:
You're right about missing `virtual` modifier. I'm adding it. But other way, `ITagDefinitionStore` declares async methods, because implementation can be changed with database, or any rest request implementation.

Also ReactionDefinitionStore uses synchronous task implementation.
https://github.com/abpframework/abp/blob/a92a5c98ab95aab4c456d9068c141e6fddba8109/modules/cms-kit/src/Volo.CmsKit.Domain/Volo/CmsKit/Reactions/DefaultReactionDefinitionStore.cs

Reply:
It looks like `virtual` is still missing.